src/components/rating/rating.tsx

```diff
@@ -238,27 +241,88 @@ export class Rating
     this.setFocus();
   }
 
-  private updateValue(value: number) {
-    this.value = value;
-    this.calciteRatingChange.emit({ value });
-  }
+  private handleRatingPointerOver = () => {
+    this.isKeyboardInteraction = false;
+  };
 
-  private onKeyboardPressed = (event: KeyboardEvent): void => {
-    if (!this.required && isActivationKey(event.key)) {
-      event.preventDefault();
-      this.updateValue(0);
-    }
+  private handleRatingPointerOut = () => {
+    this.isKeyboardInteraction = true;
+    this.hoverValue = null;
+    this.focusValue = null;
+  };
+
+  private handleRatingFocusLeave = (): void => {
+    this.focusValue = null;
+    this.hasFocus = false;
+    this.hoverValue = null;
   };
 
-  private onFocusChange = (selectedRatingValue: number): void => {
+  private handleRatingFocusIn = (): void => {
+    this.focusValue = this.value > 0 ? this.value : 1;
     this.hasFocus = true;
-    if (!this.required && this.focusValue === selectedRatingValue) {
-      this.updateValue(0);
+    this.hoverValue = this.value > 0 ? this.value : 1;
+  };
+
+  private handleLabelPointerOver = (ev: PointerEvent) => {
+    const newPointerValue = ev.currentTarget["nextElementSibling"]["value"] || this.value || 0;
+    this.hoverValue = newPointerValue;
+    //  when switching from keyboard to mouse this prop change is required to remove the focus styles from the element which are needed when navigating with a keyboard.
+    this.focusValue = null;
+  };
+
+  private handleKeyDown = (ev: KeyboardEvent) => {
+    if (ev.target["nodeName"] === "CALCITE-RATING") {
+      this.isKeyboardInteraction = true;
     } else {
-      this.focusValue = selectedRatingValue;
+      const inputVal = Number(ev.target["value"]);
```

Comment:
I did try replacing ev.target with inputFocusRef. this creates a circular dependency error because inputFocusRef relies on this.value to understand it's current context.